src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -2044,6 +2044,8 @@ def create_sync_policy_group(self, group_id: str, status: str, bucket_name: str
                                          http_status_code=500, component='rgw')
         except SubprocessError as error:
             raise DashboardException(error, http_status_code=500, component='rgw')
+        if not bucket_name:
```

Comment:
let's also introduce a parameter to decide whether it needs a period commit. The reason is we have a method called `create_dashboard_admin_sync_group` which creates the policy group for a bucket and it also does a period commit after all the policies are created rather than every time (as you came to realize period commit is expensive when you have cpu limitations). 
```suggestion
        if not bucket_name and update_period:
```

Reply:
If there is no bucket_name provided then it means we are making changes in zonegroup level policy, and every zonegroup level policy changes needs a period update, as per the doc.

Reply:
yup, it does. but the point i was raising is that `create_dashboard_admin_sync_group` is already running a period commit after everything gets created so that it won't cause any timeouts in "dev" environments. so we need a condition to handle whether it needs a period update or not so that the `self.create_sync_policy_group` and other calls won't call period update multiple times, rather it calls it one time only. You can even make the new `update_period` param True by default but that should be false in the function I put below.
https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/services/rgw_client.py#L2229

Reply:
okay, got your point! Thanks for clarification